packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/v21-sign-transaction-endpoint.test.ts

```diff
@@ -37,164 +38,168 @@ import { PluginRegistry } from "@hyperledger/cactus-core";
-  const web3 = new Web3(web3Provider);
-  const web3JsQuorum: IWeb3Quorum = Web3JsQuorum(web3);
-
-  const orionKeyPair = await besuTestLedger.getOrionKeyPair();
-  const besuKeyPair = await besuTestLedger.getBesuKeyPair();
-
-  const besuPrivateKey = besuKeyPair.privateKey.toLowerCase().startsWith("0x")
-    ? besuKeyPair.privateKey.substring(2)
-    : besuKeyPair.privateKey; // besu node's private key
-
-  const contractOptions = {
-    data: `0x123`,
-    // privateFrom : Orion public key of the sender.
-    privateFrom: orionKeyPair.publicKey,
-    // privateFor : Orion public keys of recipients or privacyGroupId: Privacy group to receive the transaction
-    privateFor: [orionKeyPair.publicKey],
-    // privateKey: Ethereum private key with which to sign the transaction.
-    privateKey: besuPrivateKey,
-  };
-
-  const transactionHash =
-    await web3JsQuorum.priv.generateAndSendRawTransaction(contractOptions);
-
-  const transaction = await web3.eth.getTransaction(transactionHash);
-  const singData = jsObjectSigner.sign(transaction.input);
-  const signDataHex = Buffer.from(singData).toString("hex");
-
-  const request: SignTransactionRequest = {
-    keychainId,
-    keychainRef,
-    transactionHash: transactionHash,
-  };
-
-  const configuration = new BesuApiClientOptions({ basePath: node1Host });
-  const api = new BesuApiClient(configuration);
-
-  // Test for 200 valid response test case
-  const res = await api.signTransactionV1(request);
-  t.ok(res, "API response object is truthy");
-  t.deepEquals(signDataHex, res.data.signature, "Signature data are equal");
-
-  // Test for 404 Transaction not found test case
-  try {
-    const notFoundRequest: SignTransactionRequest = {
-      keychainId: "fake",
-      keychainRef: "fake",
-      transactionHash:
-        "0x46eac4d1d1ff81837698cbab38862a428ddf042f92855a72010de2771a7b704d",
+  afterAll(async () => {
+    await apiServer.shutdown();
+  });
+
+  test(testCase, async () => {
+    const keyEncoder: KeyEncoder = new KeyEncoder("secp256k1");
+    const keychainId = uuidv4();
+    const keychainRef = uuidv4();
+
+    const { privateKey } = Secp256k1Keys.generateKeyPairsBuffer();
+    const keyHex = privateKey.toString("hex");
+    const pem = keyEncoder.encodePrivate(keyHex, KeyFormat.Raw, KeyFormat.PEM);
+
+    const keychain = new PluginKeychainMemory({
+      backend: new Map([[keychainRef, pem]]),
+      keychainId,
+      logLevel,
+      instanceId: uuidv4(),
+    });
+
+    const httpServer1 = createServer();
+    await new Promise((resolve, reject) => {
+      httpServer1.once("error", reject);
+      httpServer1.once("listening", resolve);
+      httpServer1.listen(0, "127.0.0.1");
+    });
+    const addressInfo1 = httpServer1.address() as AddressInfo;
+    log.debug(`HttpServer1 AddressInfo: ${JSON.stringify(addressInfo1)}`);
+
+    const node1Host = `http://${addressInfo1.address}:${addressInfo1.port}`;
+    log.debug(`Cactus Node 1 Host: ${node1Host}`);
```

Comment:
Please remove `1` from `httpServer1`, `addressInfo1` and `node1Host`. I see it was named that way in the original test but we can fix it now anyway